docs/tutorials/cwp_calib_tutorial.rst

```diff
@@ -0,0 +1,185 @@
+===========================================================
+Borehole Calibration with Selective Simulation Cancellation
+===========================================================
+
+Introduction - Calibration with libEnsemble and CWP
+---------------------------------------------------
+
+This tutorial demonstrates libEnsemble's capability to selectively cancel pending
+simulations based on instructions from the *Persistent CWP* calibration
+generator function. This capability is critical for this calibration use-case since
+it isn't useful for the generator to receive extraneous evaluations
+from resources that may be more effectively applied towards critical evaluations.
+
+[JLN: BETTER JUSTIFICATION GOES HERE?]
+
+For a somewhat different approach than libEnsemble's :doc:`other tutorials<tutorials>`,
+we'll emphasize the settings, functions, and data fields within the calling script, CWP
+:ref:`persistent generator<persistent-gens>`, Manager, and :ref:`sim_f<api_sim_f>`
+that make this capability possible, rather than outlining a step-by-step process
+for writing this exact use-case.
+
+.. note::
+    The generator function featured in this tutorial can be found in ``gen_funcs/persistent_cwp_calib.py``. This version uses simplified standalone routines in place of the in-development CWP library.
+
+Generator - Point Cancellation Requests and Dedicated Fields
+------------------------------------------------------------
+
+Given "observed values" at a given set of points called "x"s, the CWP generator seeks to fit
+a Gaussian process model to these points using a function parameterized with
+"Thetas". The goal is to find the Theta that most closely matches observed values.
+
+After an initial batch of randomly sampled values, the model generates
+new Thetas. Each Theta is evaluated via the ``sim_f`` at each of the points, until
+some error threshold is reached.
+
+[JLN: BETTER DESCRIPTION OF PROBLEM GOES HERE?]
+
+While the generator loops and updates the model based on returned
+points from simulations, it detects using a library function if any pending points
+and Thetas distributed for simulation are no longer needed for the model
+and ought to be cancelled (obviated). The generator then calls ``cancel_row()``::
+
+    r_obviate = obviate_pend_thetas(model, theta, data_status)
+    if r_obviate[0].shape[0] > 0:
+        cancel_row(pre_count, r_obviate, n_x, data_status, comm)
+
+``pre_count`` is a matrix of Thetas and "x"s, ``r_obviate`` is a selection
```

Comment:
pre_count is a scalar of the number of sim_ids to used for the test values, basically an offset.